File: backend/QuizHubApplication/DTOs/Responses/UserQuizAttemptQuestionDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuizHubApplication.DTOs.Responses
{
    public record UserQuizAttemptQuestionDto(int Id, string Text, int Points, string QuestionType, List<UserQuizAttemptOptionDto> OptionDtos)
    {
    }
}

File: frontend/QuizHubFrontend/src/app/features/quiz-attempts/model/UserQuizAttemptQuestionDto.ts
import { UserQuizAttemptOptionDto } from './UserQuizAttemptOptionDto';

export interface UserQuizAttemptQuestionDto {
    id: number;
    text: string;
    points: number;
    questionType: string;
    answered: boolean;
    optionDtos: UserQuizAttemptOptionDto[];
}
